src/components/Router/useWorkflowSteps.tsx

```diff
@@ -0,0 +1,203 @@
+import { StepConfig } from '~types/steps'
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks'
+
+import { formatStep } from '../../index'
+import { NarrowSdkOptions, UrlsConfig } from '~types/commons'
+import {
+  StepsLoadingStatus,
+  StepsHook,
+  CompleteStepValue,
+} from '~types/routers'
+import { poller, PollFunc, Engine } from '../WorkflowEngine'
+import type { WorkflowResponse } from '../WorkflowEngine/utils/WorkflowTypes'
+import useUserConsent from '~contexts/useUserConsent'
+
+type WorkflowStepsState = {
+  status: StepsLoadingStatus
+  steps: StepConfig[]
+  taskId: string | undefined
+  error: string | undefined
+}
+
+const defaultState: WorkflowStepsState = {
+  status: 'idle',
+  taskId: undefined,
+  error: undefined,
+  steps: [],
+}
+
+export const createWorkflowSteps = (
+  { token, workflowRunId, ...options }: NarrowSdkOptions,
+  { onfido_api_url }: UrlsConfig
+): StepsHook => () => {
+  const { addUserConsentStep } = useUserConsent()
+
+  //@todo: We should move the logic of enforcing the `welcome` step from index.ts here
+  const [state, setState] = useState<WorkflowStepsState>({
+    ...defaultState,
+    steps: options.steps,
+  })
+
+  useEffect(() => {
+    // We only inject this step in the first workflow task
+    if (options.steps.every(({ type }) => type !== 'welcome')) {
```

Comment:
this is assuming that there is always a welcome step. We should have a different mechanism to figure out, if this is the 1st workflow task

Reply:
And also that's why I added a @todo, to group all the code that set the lists of steps in the dedicated 'use*Steps'.

Reply:
>For now there is always welcome Step as default first step. To initiate the SDK with that. This may require future product spec and re-engineering the SDK

This only applies to workflow steps though, right?